adapters/network.js

```diff
@@ -161,23 +162,76 @@ const httpPATCH = async (url, data, options) => {
   return clientResponse;
 };
 
-/**
- * Prepares the proxy request
- * @param {*} request
- * @returns
- */
-const prepareProxyRequest = request => {
-  const { body, method, params, endpoint } = request;
-  let { headers } = request;
-  let data;
+const getPayloadData = body => {
   let payload;
   let payloadFormat;
-  for (const [key, value] of Object.entries(body)) {
+  Object.entries(body).forEach(([key, value]) => {
     if (!_.isEmpty(value)) {
       payload = value;
       payloadFormat = key;
     }
+  });
+  return { payload, payloadFormat };
+};
+
+/**
+ * Method for stringification of query parameters
+ * To understand the use-case for this method, please take a look at the code mentioned below:
+ *
+ * @param {*} value
+ * @returns {String}
+ */
+function stringifyQueryParam(value) {
+  let stringifiedValue = `${value}`;
+  if (Array.isArray(value)) {
+    stringifiedValue = value.map(v => stringifyQueryParam(v)).join(",");
+    return `[${stringifiedValue}]`;
+  }
+  if (value && typeof value === "object") {
+    // check for value is being done to avoid null inside since typeof null = "object"
+    stringifiedValue = JSON.stringify(value);
+  }
+  return stringifiedValue;
+}
+
+/**
+ * Obtain FORM payload-format data to send the data to destination
+ *
+ * @param {Object} payload
+ * @returns {String}
+ */
+function getFormData(payload, isCustom) {
+  let data = new URLSearchParams();
+  Object.keys(payload).forEach(key => {
+    let payloadValStr = `${payload[key]}`;
+    if (typeof payload[key] === "object") {
+      payloadValStr = stringifyQueryParam(payload[key]);
+    }
+    data.append(key, payloadValStr);
+  });
+  if (!isCustom) {
```

Comment:
If these are unavoidable changes needed for comparison, they should be moved to the `proxyTest` controller.